aten/src/ATen/cpu/vec/vec512/vec512_bfloat16.h

```diff
@@ -1401,6 +1425,30 @@ inline void transpose_mxn<Half, 32, 32>(
   }
 }
 
+template <typename T, int M, int N,
+          typename std::enable_if_t<std::is_same<T, Half>::value && ((M < 32 && M != 16) || (N < 32 && N != 16)), int> = 0>
+inline void transpose_mxn(const Half* src, int64_t ld_src, Half* dst, int64_t ld_dst) {
+  // load from src
+  __mmask32 src_mask = (1 << N) - 1;
```

Comment:
Looks like same implementation between `Half` and `BFloat16`. Can we move it into a common util function?

Reply:
We must specialize this function here to ensure the priority of this function.

Reply:
We can still provide a util function as
`
template <typename T, typename std::enable_if_t<std::is_same_v<T, Half> || std::is_same_v<T, BF16>>>
transpose_mxn_32_32(const T* src, int64_t ld_src, T* dst, int64_t ld_dst)
`
then reuse this util function for BF16 and FP16, it can save lines of code.